lib/Doctrine/ODM/PHPCR/Mapping/ClassMetadata.php

```diff
@@ -329,6 +329,13 @@ class ClassMetadata implements ClassMetadataInterface
     public $parentClasses = array();
 
     /**
+     * READ-ONLY: Child class restrictions.
+     *
+     * @var array
+     */
+    public $childRestrictions = null;
```

Comment:
attribute is better imho, now that we decided it can be simply a list of
allowed classes